optimum/intel/openvino/modeling_timm.py

```diff
@@ -0,0 +1,398 @@
+import os
+from collections import OrderedDict
+from typing import Dict, List, Optional, Tuple, Union
+
+import numpy as np
+import timm
+import torch
+from timm.layers.config import set_fused_attn
+from timm.models._hub import load_model_config_from_hf
+from torch.nn import BCEWithLogitsLoss, CrossEntropyLoss, MSELoss
+from transformers import PretrainedConfig, PreTrainedModel
+from transformers.image_processing_utils import BaseImageProcessor, BatchFeature, get_size_dict
+from transformers.image_transforms import resize, to_channel_dimension_format
+from transformers.image_utils import (
+    IMAGENET_STANDARD_MEAN,
+    IMAGENET_STANDARD_STD,
+    ChannelDimension,
+    ImageFeatureExtractionMixin,
+    ImageInput,
+    PILImageResampling,
+    make_list_of_images,
+    to_numpy_array,
+    valid_images,
+)
+from transformers.modeling_outputs import BaseModelOutput, ImageClassifierOutput
+from transformers.utils import TensorType
+
+from optimum.exporters.onnx.model_configs import ViTOnnxConfig
+
+
+set_fused_attn(False, False)
+
+
+class TimmConfig(PretrainedConfig):
+    model_type = "timm"
+
+    @classmethod
+    def from_pretrained(
+        cls,
+        pretrained_model_name_or_path: Union[str, os.PathLike],
+        cache_dir: Optional[Union[str, os.PathLike]] = None,
+        force_download: bool = False,
+        local_files_only: bool = False,
+        token: Optional[Union[str, bool]] = None,
+        revision: str = "main",
+        **kwargs,
+    ) -> "PretrainedConfig":
+        kwargs["cache_dir"] = cache_dir
+        kwargs["force_download"] = force_download
+        kwargs["local_files_only"] = local_files_only
+        kwargs["revision"] = revision
+
+        config_dict = load_model_config_from_hf(pretrained_model_name_or_path)[0]
+        config_dict["num_labels"] = config_dict.pop("num_classes")
+        config_dict["image_size"] = config_dict.get("input_size")[-1]
+
+        return cls.from_dict(config_dict, **kwargs)
+
+
+class TimmOnnxConfig(ViTOnnxConfig):
+    DEFAULT_TIMM_ONNX_OPSET = 13
+    outputs = OrderedDict([("logits", {0: "batch_size"})])
+
+
+class TimmPreTrainedModel(PreTrainedModel):
+    config_class = TimmConfig
+    base_model_prefix = "timm"
+    main_input_name = "pixel_values"
+
+
+class TimmModel(TimmPreTrainedModel):
```

Comment:
I don't think we need such a model hierarchy, with TimmPreTrainedModel, TimmModel, etc. At least Optimum-Intel is not the place for such classes. What is required is just to put the logic of model creation and prefetch into the .from_timm() method of OVModelForTimm. Moreover, OVModelForTimm itself is not needed since there is no intent to expose it. It can be just a utility function or class that is not inherited from HF hierarchy at all.

Reply:
I have removed the hierarchy as much as possible, leaving a small placeholder model to contain the `timm` models.